### PR TITLE
Fix: add PolicySelect component to support select the policy in deploy workflow step

### LINKED ISSUE #499: Add PolicySelect component to support select the policy in deploy workflow step
List the policies with the app name, and show options.

### PR DESCRIPTION
**Pull request by wangbow**
### Description of your changes

Fixes #499 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [x] Run `yarn lint` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: src/interface/application.ts [hunk 1]
```diff
@@ -382,3 +382,13 @@ export interface ComponentDefinitionsBase {
   name: string;
   workloadType?: string;
 }
+
+export interface PoliciesBase {
```
Comment (barnettZQG) [inline src/interface/application.ts:386]:
```suggestion
export interface PolicyBase {
```

File: src/extends/PolicySelect/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import locale from '../../utils/locale';
+import i18n from '../../i18n';
+
+type Props = {
+  onChange: (value: any) => void;
+  policySelectDataSource?: string[];
+  value: any;
+  id: string;
+  disabled: boolean;
+  appName: string;
```
Comment (barnettZQG) [inline src/extends/PolicySelect/index.tsx:14]: We should get the app name from the `uischema` model in the dva, so we should save the app name to the model before rendering the  UISchema component in the application layout.

File: src/extends/PolicySelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  };
+
+  fetchPolicyList = async () => {
+    const { appName = '' } = this.props;
+    getPolicyList({
+      name: appName,
+    })
+      .then((res) => {
+        if (res && res.policies) {
+          const policyListData = (res.policies || []).map((item: PoliciesBase) => item.name);
```
Comment (barnettZQG) [inline src/extends/PolicySelect/index.tsx:40]: We should show the policy with `name(type)` format

File: src/extends/PolicySelect/index.tsx (hunk 2 above)
Comment (wangbow) [reply src/extends/PolicySelect/index.tsx:14]: Good idea, let me try it

File: src/interface/application.ts (hunk 1 above)
Comment (wangbow) [reply src/interface/application.ts]: done

File: src/extends/PolicySelect/index.tsx (hunk 2 above)
Comment (wangbow) [reply src/extends/PolicySelect/index.tsx]: done

File: src/extends/PolicySelect/index.tsx (hunk 3 above)
Comment (wangbow) [reply src/extends/PolicySelect/index.tsx]: done

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**